swarm/swarm.go

```diff
@@ -51,19 +58,32 @@ func run(cmd string) {
 }
 
 func getleaderHost() string {
-	var response []Response
-	var host string
+	var nodeStats NodeStats
 	client := &http.Client{}
-	resp, _ := client.Get("http://" + os.Getenv("HOST") + ":7001/v2/admin/machines")
-	defer resp.Body.Close()
+	resp, _ := client.Get("http://" + os.Getenv("HOST") + ":2379/v2/stats/self")
+
 	body, _ := ioutil.ReadAll(resp.Body)
-	json.Unmarshal(body, &response)
-	for _, node := range response {
-		if node.State == "leader" {
-			host = strings.Split(node.ClientURL, "//")[1]
+	json.Unmarshal(body, &nodeStats)
+
+	etcdLeaderID := nodeStats.LeaderInfo.Name
+
+	var etcdCluster EtcdCluster
+	resp, _ = client.Get("http://" + os.Getenv("HOST") + ":2379/v2/members")
+	defer resp.Body.Close()
+
+	body, _ = ioutil.ReadAll(resp.Body)
+	json.Unmarshal(body, &etcdCluster)
+
+	for _, node := range etcdCluster.Members {
+		if node.ID == etcdLeaderID {
+			u, err := url.Parse(node.ClientURLs[0])
+			if err == nil {
```

Comment:
should we be throwing an error in this case? Discarding the error sounds like we could run into trouble later on when we can't figure out why it's not booting.